File: ruby/helpers_spec.rb
require 'minitest/autorun'
require_relative 'helpers'

include ProjectEuler

describe "Helpers" do
  include Helpers

  describe "fibonacci" do
    it "should return the nth fibonacci number" do
      fibonacci(9).must_equal 34
    end
  end

  describe "fibonacci_sequence" do
    it "should return an array of fibonacci numbers whose values do not exceed n" do
      fibonacci_sequence(10).must_equal [0, 1, 2, 3, 5, 8]
    end
  end

  describe "prime_numbers" do
    it "should return an array of prime numbers up to n" do
      prime_numbers(19).must_equal [2, 3, 5, 7, 11, 13, 17, 19]
    end
  end

  describe "prime_factors" do
    it "should return an array of prime factors for a given number" do
      prime_factors(13_195).must_equal [5, 7, 13, 29]
    end
  end
end

describe "FixnumExtension" do
  Fixnum.send :include, FixnumExtensions

  describe "palindrome?" do
    it "should return true if the given number is a palindrome" do
      9009.palindrome?.must_equal true
    end

    it "should return false if the given number is not a palindrome" do
      1234.palindrome?.must_equal false
    end
  end
end

File: haskell/HelpersSpec.hs
module HelpersSpec where

import Test.Hspec
import Helpers

main :: IO ()
main = hspec $ do
  describe "fibs" $ do
    it "should return a list of fibonacci numbers" $
      [x | x <- takeWhile (< 100) fibs] `shouldBe` [1,1,2,3,5,8,13,21,34,55,89]

  describe "primes" $ do
    it "should return a list of prime numbers" $
      [x | x <- takeWhile (< 50) primes]
      `shouldBe`
      [2,3,5,7,11,13,17,19,23,29,31,37,41,43,47]

  describe "primeFactors" $ do
    it "should return a list of prime factors for a given number" $
      primeFactors 13195 `shouldBe` [5,7,13,29] 

  describe "palindrome" $ do
    it "should return true if the given number is a palindrome" $
      (palindrome . show) 9009 `shouldBe` True

    it "should return false if the given number is not a palindrome" $
      (palindrome . show) 1234 `shouldBe` False
